src/onepasswordconnectsdk/client.py

```diff
@@ -381,19 +383,25 @@ def sanitize_for_serialization(self, obj):
         return self.serializer.sanitize_for_serialization(obj)
 
 
-def new_client(url: str, token: str, is_async: bool = False) -> Union[AsyncClient, Client]:
+def new_client(url: str, token: str, is_async: bool = False, certificate: str = None) -> Union[AsyncClient, Client]:
     """Builds a new client for interacting with 1Password Connect
     Parameters:
     url: The url of the 1Password Connect API
     token: The 1Password Service Account token
     is_async: Initialize async or sync client
+    certificate: Optional path to custom certificate bundle for verification
 
     Returns:
     Client: The 1Password Connect client
     """
-    if is_async:
+    if is_async and certificate:
+        return AsyncClient(url, token, certificate)
+    elif is_async:
         return AsyncClient(url, token)
-    return Client(url, token)
+    elif certificate:
+        return Client(url, token, certificate)
+    else:
+        return Client(url, token)
```

Comment:
As the constructor params starts to grow, to avoid growing the amount of the params more in the future, I think it would be good to pass `clientConfig: ClientConfig` instead of passing a specific `certificate` param.

So, the idea is that `ClientConfig` class will have the `certificate: str` param, and in the future we can easily extend it with more, if we need them. In addition to that, we can make `ClientConfig` inherit the `httpx.Client` class, so we will be able to pass all the properties that `httpx.Client` accepts, and not only `certificate`.

Does it make sense?

Reply:
That does make sense. I just did a search of this repo and don't see a "ClientConfig". That would be a new class created?

Reply:
Yes, that should be a new class. Ideally, having the same parameters as in `httpx.Client` and `httpx.AsyncClient` classes.